sound/soc/sof/intel/hda-loader.c

```diff
@@ -519,14 +519,16 @@ int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
 			      struct sof_ipc4_fw_library *fw_lib, bool reload)
 {
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
 	struct hdac_ext_stream *hext_stream;
 	struct firmware stripped_firmware;
 	struct sof_ipc4_msg msg = {};
 	struct snd_dma_buffer dmab;
 	int ret, ret1;
 
-	/* IMR booting will restore the libraries as well, skip the loading */
-	if (reload && hda->booted_from_imr)
+	/* if IMR booting is enabled and library reload is not defined, skip the loading */
+	if (reload && hda->booted_from_imr &&
+	    !(ipc4_data->fw_image_flags & SOF_IPC4_LIBRARY_RELOAD))
```

Comment:
I would opt for a more generic name still, like `SOF_IPC4_CONTEXT_NOT_PRESERVED` or something (probably toss in _D3_ to _IMR_? or just use comment to clarify it's meaning and keeping it generic).
If we ever want or need to experiment with the context save/restore at large in other parts of the code.

Reply:
@ujfalusi I use the name SOF_IPC4_FW_CONTEXT_LOST, how about it ? Thanks!